### ISSUE TITLE
[F-A-5] LoRA Fine-Tune - Llama-3-8B Model Training

### ISSUE THREAD

**Issue by siven76** [siven76/PA-AI#7]
## Feature Description
Implement LoRA (Low-Rank Adaptation) fine-tuning pipeline for Llama-3-8B-Instruct model using Q&A pairs from F-A-4 to create a specialized price action knowledge assistant.

## Acceptance Criteria
- [ ] Runs on single RTX 4090 (24 GB VRAM)
- [ ] Training completes within 3 epochs maximum
- [ ] Evaluation loss ≤ 10% above training loss
- [ ] Memory-efficient training with gradient checkpointing
- [ ] Model checkpointing and resumable training
- [ ] Validation on held-out test set
- [ ] Export trained LoRA adapters
- [ ] Integration with inference pipeline

## Technical Requirements
- **Base Model**: Meta Llama-3-8B-Instruct
- **Training Method**: QLoRA (4-bit quantization + LoRA)
- **Framework**: HuggingFace PEFT + TRL
- **Hardware**: Single RTX 4090 (24GB VRAM)
- **Memory Optimization**: Gradient checkpointing, mixed precision
- **Data Format**: JSONL from F-A-4 SFT Data Builder

## Training Configuration
```yaml
model:
  base_model: "meta-llama/Llama-3-8b-instruct"
  quantization: "4bit"
  
lora:
  r: 16
  alpha: 32
  dropout: 0.1
  target_modules: ["q_proj", "v_proj", "k_proj", "o_proj"]

training:
  batch_size: 4
  gradient_accumulation_steps: 4
  learning_rate: 2e-4
  max_epochs: 3
  warmup_ratio: 0.1
  weight_decay: 0.01
  
optimization:
  gradient_checkpointing: true
  mixed_precision: "bf16"
  dataloader_num_workers: 4
```

## Performance Targets
- **Training Time**: < 8 hours for full dataset
- **Memory Usage**: < 22 GB VRAM peak
- **Convergence**: Eval loss plateau within 3 epochs
- **Model Size**: LoRA adapters < 100 MB
- **Inference Speed**: Compatible with F-A-6 latency requirements

## Validation Metrics
- Perplexity on validation set
- BLEU score for answer quality
- Citation accuracy (source attribution)
- Factual consistency with source material
- Response relevance to questions

## Definition of Done
- [ ] Training pipeline with QLoRA implementation
- [ ] Automated hyperparameter validation
- [ ] Model checkpointing and recovery
- [ ] Evaluation metrics and reporting
- [ ] LoRA adapter export and versioning
- [ ] Integration tests with inference API
- [ ] Training documentation and runbooks
- [ ] Performance benchmarks and optimization guide
